src/Servers/Kestrel/Core/src/Internal/Infrastructure/TimingPipeFlusher.cs

```diff
@@ -111,5 +115,11 @@ private async ValueTask<FlushResult> TimeFlushAsync(MinDataRate minRate, long co
 
             return default;
         }
+
+        private async Task<FlushResult> AwaitLastFlushAndTimeFlushAsync(Task lastFlushTask, MinDataRate minRate, long count, IHttpOutputAborter outputAborter, CancellationToken cancellationToken)
```

Comment:
Any reason not `ValueTask<FlushResult>`? `Task<FlushResult>` will always allocate.

Reply:
In this case we know the lastFlushTask is not complete so we'll allocate anyway. And it removes some calls to AsTask().